### PR TITLE
add a static popup to confirm deletion of auras

### PR DESCRIPTION
**Pull request by emptyrivers**
Implements [wowace issue 644](https://www.wowace.com/projects/weakauras-2/issues/644).

### CODE REVIEW

**Comment by InfusOnWoW** [comment]
Modern UI Design shuns message boxes for Delete confirmation. 

It has been replaced by undo, either limited to one action: For example, Windows's trashcan for deleted files, or how e.g. gmail handles deleted emails. 

Or in other applications by just being part of a global undo & redo.

I actually had a half-finished patch for undoing deletion at one point, but the user interface didn't work out, and it didn't feel that great, that's why we didn't take it on.

So, we can probably merge this (after a review), but leave the ticket open

**Comment by Stanzilla** [comment]
If you touch an ace widget you have to bump the minor version at the top of the file

File: WeakAurasOptions/WeakAurasOptions.lua [hunk 1]
```diff
@@ -1256,6 +1240,47 @@ function WeakAuras.DeleteOption(data)
...
+    self.data = nil
+  end,
+  showAlert = true,
+  whileDead = true,
+  preferredindex = STATICPOPUP_NUMDIALOGS,
+}
+
+function WeakAuras.ConfirmDelete(toDelete, parents)
+  if toDelete then
+    StaticPopup_Show("WEAKAURAS_CONFIRM_DELETE", "", "", {toDelete = toDelete, parents = parents})
```
Comment (InfusOnWoW) [inline WeakAurasOptions/WeakAurasOptions.lua]: The message should tell the user what they are deleting, that this the group name, aura name, or for any other case properly somthing like "deleting x auras". 

The buttons in message boxes should never be "yes"/"no", but always verbs indicating the action that is about to be taken.

In this case: "Keep" or "Cancel" and "Delete".

**Comment by emptyrivers** [comment]
I have improved the warning text.

File: WeakAurasOptions/WeakAurasOptions.lua [hunk 2]
```diff
@@ -1256,6 +1240,49 @@ function WeakAuras.DeleteOption(data)
...
+}
+
+function WeakAuras.ConfirmDelete(toDelete, parents)
+  if toDelete then
+    local warningForm = L["You are about to delete %d aura(s). |cFFFF0000This cannot be undone!|r Would you like to continue?"]
+    StaticPopupDialogs["WEAKAURAS_CONFIRM_DELETE"].text = warningForm:format(#toDelete)
+    StaticPopup_Show("WEAKAURAS_CONFIRM_DELETE", "", "", {toDelete = toDelete, parents = parents})
+  end
+end
+
```
Comment (Stanzilla) [inline WeakAurasOptions/WeakAurasOptions.lua]: lgtm minus those newlines here

**Pull request merged**